backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteNonRootBranch.java

```diff
@@ -38,6 +40,11 @@ public GitMacheteNonRootBranch(String name,
       ISyncToRemoteStatus syncToRemoteStatus,
       SyncToParentStatus syncToParentStatus,
       @Nullable String customAnnotation) {
+    log.debug(
```

Comment:
As discussed, this should be optimized wrt. performance to avoid rendering the string when DEBUG logging is off; either wrap the string into a nullary lambda (if SLF4J allows for any lazy evaluation like that), or use SLF4J's own formatting.

Reply:
Maybe some of the stuff discussed here would be useful https://stackoverflow.com/questions/41255503/lambda-support-for-slf4j-api ? Coz generally interpolation rocks and the classic SL4FJ formatting will suck a big time for such long strings. I'm afraid though that the suggested solutions (like using SLF4J 2.0.0) might be make it impossible to get the classpath right :/

Reply:
https://mvnrepository.com/artifact/kr.pe.kwonnam.slf4j-lambda actually looks pretty ok despite having few stars on GH... as a last resort, you can always implement a similar functionality in a new `logging` subproject

Reply:
OK, so I'm using `slf4j-lambda` for lazy evaluation + `better-strings` for string interpolations